### ISSUE TITLE
iBus 1.5.19 compile error from Gentoo emerge

### ISSUE THREAD

**Issue by K73SK** [ibus/ibus#2083]
Please fill in the following items if you don't know the root cause.

Which distribution and version?: 
4.19.23-gentoo

Which desktop environment and version?: 
KDE

Which session type?: 
X11

Which application and version?: 
N/A

IBus version?: 
1.5.19

Issue description:
Please see the bottom of this message for the emerge output. When I do an update on ibus, I get a compile error. It appears some bad code was pushed out to the repo or something. Related to `emojier.c`

Steps to reproduce:
1. emerge app-i18n/ibus

Can you reproduce your problem when you restart ibus-daemon? (yes / no): 
N/A

Do you see any errors when you run ibus-daemon with the verbose option?:
N/A

Can you reproduce your problem with a new user account instead of the current your account? (yes / no): 
N/A

```
emojier.c:2958:2: note: in expansion of macro ‘g_return_val_if_fail’
  g_return_val_if_fail (_tmp22_ >= 0, FALSE);
  ^~~~~~~~~~~~~~~~~~~~
emojier.c:2890:13: note: declared here
 static void ibus_emojier_clamp_page (IBusEmojier* self) {
             ^~~~~~~~~~~~~~~~~~~~~~~
emojier.c: In function ‘ibus_emojier_show_unicode_blocks’:
emojier.c:3455:19: error: invalid suffix "_" on integer constant
  g_return_if_fail 1_ = self->priv->m_category_active_index;
                   ^~
emojier.c:3455:2: error: ‘g_return_if_fail’ undeclared (first use in this function); did you mean ‘g_test_fail’?
  g_return_if_fail 1_ = self->priv->m_category_active_index;
  ^~~~~~~~~~~~~~~~
  g_test_fail
emojier.c:3455:2: note: each undeclared identifier is reported only once for each function it appears in
emojier.c:3455:18: error: expected ‘;’ before numeric constant
  g_return_if_fail 1_ = self->priv->m_category_active_index;
                  ^~~
                  ;
emojier.c:3519:83: error: ‘___lambda22__gtk_widget_button_press_event’ undeclared (first use in this function); did you mean ‘___lambda57__gtk_widget_button_press_event’?
  g_signal_connect_object ((GtkWidget*) _tmp22_, "button-press-event", (GCallback) ___lambda22__gtk_widget_button_press_event, self, 0);
                                                                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                                                                                   ___lambda57__gtk_widget_button_press_event
emojier.c:3548:79: error: ‘___lambda23__gtk_list_box_row_activated’ undeclared (first use in this function); did you mean ‘___lambda58__gtk_list_box_row_activated’?
  g_signal_connect_object ((GtkListBox*) _tmp37_, "row-activated", (GCallback) ___lambda23__gtk_list_box_row_activated, self, 0);
                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                                                                               ___lambda58__gtk_list_box_row_activated
emojier.c:3642:78: error: ‘___lambda24__gtk_widget_size_allocate’ undeclared (first use in this function); did you mean ‘___lambda59__gtk_widget_size_allocate’?
  g_signal_connect_object ((GtkWidget*) _tmp69_, "size-allocate", (GCallback) ___lambda24__gtk_widget_size_allocate, self, 0);
                                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                                                                              ___lambda59__gtk_widget_size_allocate
emojier.c: At top level:
emojier.c:3776:8: error: unknown type name ‘Block1Data’
 static Block1Data* block1_data_ref (Block1Data* _data1_) {
        ^~~~~~~~~~
emojier.c:3776:37: error: unknown type name ‘Block1Data’; did you mean ‘Block5Data’?
 static Block1Data* block1_data_ref (Block1Data* _data1_) {
                                     ^~~~~~~~~~
                                     Block5Data
emojier.c: In function ‘block1_data_unref’:
emojier.c:3783:2: error: unknown type name ‘Block1Data’; did you mean ‘Block5Data’?
  Block1Data* _data1_;
  ^~~~~~~~~~
  Block5Data
emojier.c:3784:13: error: ‘Block1Data’ undeclared (first use in this function); did you mean ‘Block5Data’?
  _data1_ = (Block1Data*) _userdata_;
             ^~~~~~~~~~
             Block5Data
emojier.c:3784:24: error: expected expression before ‘)’ token
  _data1_ = (Block1Data*) _userdata_;
                        ^
In file included from /usr/lib64/glib-2.0/include/glibconfig.h:9,
                 from /usr/include/glib-2.0/glib/gtypes.h:32,
                 from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from emojier.c:26:
emojier.c:3785:41: error: request for member ‘_ref_count_’ in something not a structure or union
  if (g_atomic_int_dec_and_test (&_data1_->_ref_count_)) {
                                         ^~
/usr/include/glib-2.0/glib/gmacros.h:241:104: note: in definition of macro ‘G_STATIC_ASSERT’
 #define G_STATIC_ASSERT(expr) typedef char G_PASTE (_GStaticAssertCompileTimeAssertion_, __COUNTER__)[(expr) ? 1 : -1] G_GNUC_UNUSED
                                                                                                        ^~~~
emojier.c:3785:6: note: in expansion of macro ‘g_atomic_int_dec_and_test’
  if (g_atomic_int_dec_and_test (&_data1_->_ref_count_)) {
      ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib/gthread.h:32,
                 from /usr/include/glib-2.0/glib/gasyncqueue.h:32,
                 from /usr/include/glib-2.0/glib.h:32,
                 from emojier.c:26:
emojier.c:3785:41: error: request for member ‘_ref_count_’ in something not a structure or union
  if (g_atomic_int_dec_and_test (&_data1_->_ref_count_)) {
                                         ^~
/usr/include/glib-2.0/glib/gatomic.h:182:19: note: in definition of macro ‘g_atomic_int_dec_and_test’
     (void) (0 ? *(atomic) ^ *(atomic) : 1);                                  \
                   ^~~~~~
emojier.c:3785:41: error: request for member ‘_ref_count_’ in something not a structure or union
  if (g_atomic_int_dec_and_test (&_data1_->_ref_count_)) {
                                         ^~
/usr/include/glib-2.0/glib/gatomic.h:182:31: note: in definition of macro ‘g_atomic_int_dec_and_test’
     (void) (0 ? *(atomic) ^ *(atomic) : 1);                                  \
                               ^~~~~~
emojier.c:3785:41: error: request for member ‘_ref_count_’ in something not a structure or union
  if (g_atomic_int_dec_and_test (&_data1_->_ref_count_)) {
                                         ^~
/usr/include/glib-2.0/glib/gatomic.h:183:28: note: in definition of macro ‘g_atomic_int_dec_and_test’
     __sync_fetch_and_sub ((atomic), 1) == 1;                                 \
                            ^~~~~~
emojier.c:3787:17: error: request for member ‘self’ in something not a structure or union
   self = _data1_->self;
                 ^~
emojier.c:3788:28: error: request for member ‘warning_button’ in something not a structure or union
   _g_object_unref0 (_data1_->warning_button);
                            ^~
emojier.c:107:33: note: in definition of macro ‘_g_object_unref0’
 #define _g_object_unref0(var) ((var == NULL) ? NULL : (var = (g_object_unref (var), NULL)))
                                 ^~~
emojier.c:3788:28: error: request for member ‘warning_button’ in something not a structure or union
   _g_object_unref0 (_data1_->warning_button);
                            ^~
emojier.c:107:56: note: in definition of macro ‘_g_object_unref0’
 #define _g_object_unref0(var) ((var == NULL) ? NULL : (var = (g_object_unref (var), NULL)))
                                                        ^~~
emojier.c:3788:28: error: request for member ‘warning_button’ in something not a structure or union
   _g_object_unref0 (_data1_->warning_button);
                            ^~
emojier.c:107:79: note: in definition of macro ‘_g_object_unref0’
 #define _g_object_unref0(var) ((var == NULL) ? NULL : (var = (g_object_unref (var), NULL)))
                                                                               ^~~
In file included from /usr/include/glib-2.0/glib.h:76,
                 from emojier.c:26:
/usr/include/glib-2.0/glib/gslice.h:66:24: error: expected expression before ‘)’ token
   else   (void) ((type*) 0 == (mem));     \
                        ^
emojier.c:3790:3: note: in expansion of macro ‘g_slice_free’
   g_slice_free (Block1Data, _data1_);
   ^~~~~~~~~~~~
emojier.c: At top level:
emojier.c:3821:27: error: unknown type name ‘Block1Data’; did you mean ‘Block5Data’?
 static void ___lambda19_ (Block1Data* _data1_) {
                           ^~~~~~~~~~
                           Block5Data
emojier.c: In function ‘____lambda19__gtk_button_clicked’:
emojier.c:3849:2: warning: implicit declaration of function ‘___lambda19_’; did you mean ‘__lambda18_’? [-Wimplicit-function-declaration]
  ___lambda19_ (self);
  ^~~~~~~~~~~~
  __lambda18_
emojier.c: In function ‘ibus_emojier_show_arrow_buttons’:
emojier.c:3854:2: error: unknown type name ‘Block1Data’; did you mean ‘Block5Data’?
  Block1Data* _data1_;
  ^~~~~~~~~~
  Block5Data
In file included from /usr/include/glib-2.0/glib.h:76,
                 from emojier.c:26:
emojier.c:3917:26: error: ‘Block1Data’ undeclared (first use in this function); did you mean ‘Block5Data’?
  _data1_ = g_slice_new0 (Block1Data);
                          ^~~~~~~~~~
/usr/include/glib-2.0/glib/gslice.h:46:35: note: in definition of macro ‘g_slice_new0’
 #define  g_slice_new0(type)     ((type*) g_slice_alloc0 (sizeof (type)))
                                   ^~~~
/usr/include/glib-2.0/glib/gslice.h:46:40: error: expected expression before ‘)’ token
 #define  g_slice_new0(type)     ((type*) g_slice_alloc0 (sizeof (type)))
                                        ^
emojier.c:3917:12: note: in expansion of macro ‘g_slice_new0’
  _data1_ = g_slice_new0 (Block1Data);
            ^~~~~~~~~~~~
emojier.c:3918:9: error: request for member ‘_ref_count_’ in something not a structure or union
  _data1_->_ref_count_ = 1;
         ^~
emojier.c:3919:9: error: request for member ‘self’ in something not a structure or union
  _data1_->self = g_object_ref (self);
         ^~
emojier.c:3983:9: error: request for member ‘warning_button’ in something not a structure or union
  _data1_->warning_button = NULL;
         ^~
emojier.c:3995:28: error: request for member ‘warning_button’ in something not a structure or union
   _g_object_unref0 (_data1_->warning_button);
                            ^~
emojier.c:107:33: note: in definition of macro ‘_g_object_unref0’
 #define _g_object_unref0(var) ((var == NULL) ? NULL : (var = (g_object_unref (var), NULL)))
                                 ^~~
emojier.c:3995:28: error: request for member ‘warning_button’ in something not a structure or union
   _g_object_unref0 (_data1_->warning_button);
                            ^~
emojier.c:107:56: note: in definition of macro ‘_g_object_unref0’
 #define _g_object_unref0(var) ((var == NULL) ? NULL : (var = (g_object_unref (var), NULL)))
                                                        ^~~
emojier.c:3995:28: error: request for member ‘warning_button’ in something not a structure or union
   _g_object_unref0 (_data1_->warning_button);
                            ^~
emojier.c:107:79: note: in definition of macro ‘_g_object_unref0’
 #define _g_object_unref0(var) ((var == NULL) ? NULL : (var = (g_object_unref (var), NULL)))
                                                                               ^~~
emojier.c:3996:10: error: request for member ‘warning_button’ in something not a structure or union
   _data1_->warning_button = _tmp33_;
          ^~
emojier.c:3997:20: error: request for member ‘warning_button’ in something not a structure or union
   _tmp34_ = _data1_->warning_button;
                    ^~
emojier.c:3999:20: error: request for member ‘warning_button’ in something not a structure or union
   _tmp35_ = _data1_->warning_button;
                    ^~
emojier.c:4005:20: error: request for member ‘warning_button’ in something not a structure or union
   _tmp38_ = _data1_->warning_button;
                    ^~
emojier.c:4007:20: error: request for member ‘warning_button’ in something not a structure or union
   _tmp39_ = _data1_->warning_button;
                    ^~
emojier.c:4008:92: warning: implicit declaration of function ‘block1_data_ref’; did you mean ‘block5_data_ref’? [-Wimplicit-function-declaration]
   g_signal_connect_data (_tmp39_, "clicked", (GCallback) ____lambda19__gtk_button_clicked, block1_data_ref (_data1_), (GClosureNotify) block1_data_unref, 0);
                                                                                            ^~~~~~~~~~~~~~~
                                                                                            block5_data_ref
emojier.c:4008:92: warning: passing argument 4 of ‘g_signal_connect_data’ makes pointer from integer without a cast [-Wint-conversion]
   g_signal_connect_data (_tmp39_, "clicked", (GCallback) ____lambda19__gtk_button_clicked, block1_data_ref (_data1_), (GClosureNotify) block1_data_unref, 0);
                                                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:28,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from emojier.c:27:
/usr/include/glib-2.0/gobject/gsignal.h:391:25: note: expected ‘gpointer’ {aka ‘void *’} but argument is of type ‘int’
             gpointer    data,
             ~~~~~~~~~~~~^~~~
emojier.c:4034:19: error: request for member ‘warning_button’ in something not a structure or union
  _tmp53_ = _data1_->warning_button;
                   ^~
emojier.c:4039:20: error: request for member ‘warning_button’ in something not a structure or union
   _tmp55_ = _data1_->warning_button;
                    ^~
emojier.c: In function ‘ibus_emojier_lookup_emojis_from_annotation’:
emojier.c:4518:17: warning: assignment to ‘GCompareFunc’ {aka ‘int (*)(const void *, const void *)’} from incompatible pointer type ‘int (*)(const char *, const char *)’ [-Wincompatible-pointer-types]
         _tmp30_ = g_strcmp0;
                 ^
emojier.c:4526:18: warning: assignment to ‘GCompareFunc’ {aka ‘int (*)(const void *, const void *)’} from incompatible pointer type ‘int (*)(const char *, const char *)’ [-Wincompatible-pointer-types]
          _tmp34_ = g_strcmp0;
                  ^
emojier.c:4553:14: warning: assignment to ‘GCompareFunc’ {aka ‘int (*)(const void *, const void *)’} from incompatible pointer type ‘int (*)(const char *, const char *)’ [-Wincompatible-pointer-types]
      _tmp39_ = g_strcmp0;
              ^
emojier.c:4619:13: warning: assignment to ‘GCompareFunc’ {aka ‘int (*)(const void *, const void *)’} from incompatible pointer type ‘int (*)(const char *, const char *)’ [-Wincompatible-pointer-types]
     _tmp57_ = g_strcmp0;
             ^
emojier.c:4695:17: warning: assignment to ‘GCompareFunc’ {aka ‘int (*)(const void *, const void *)’} from incompatible pointer type ‘int (*)(const char *, const char *)’ [-Wincompatible-pointer-types]
         _tmp77_ = g_strcmp0;
                 ^
emojier.c:4703:18: warning: assignment to ‘GCompareFunc’ {aka ‘int (*)(const void *, const void *)’} from incompatible pointer type ‘int (*)(const char *, const char *)’ [-Wincompatible-pointer-types]
          _tmp81_ = g_strcmp0;
                  ^
emojier.c:4731:14: warning: assignment to ‘GCompareFunc’ {aka ‘int (*)(const void *, const void *)’} from incompatible pointer type ‘int (*)(const char *, const char *)’ [-Wincompatible-pointer-types]
      _tmp86_ = g_strcmp0;
              ^
emojier.c: In function ‘__lambda9_’:
emojier.c:4977:9: warning: assignment to ‘GCompareFunc’ {aka ‘int (*)(const void *, const void *)’} from incompatible pointer type ‘int (*)(const char *, const char *)’ [-Wincompatible-pointer-types]
  _tmp6_ = g_strcmp0;
         ^
emojier.c: At top level:
emojier.c:5210:8: error: unknown type name ‘Block2Data’
 static Block2Data* block2_data_ref (Block2Data* _data2_) {
        ^~~~~~~~~~
emojier.c:5210:37: error: unknown type name ‘Block2Data’; did you mean ‘Block5Data’?
 static Block2Data* block2_data_ref (Block2Data* _data2_) {
                                     ^~~~~~~~~~
                                     Block5Data
emojier.c: In function ‘block2_data_unref’:
emojier.c:5217:2: error: unknown type name ‘Block2Data’; did you mean ‘Block5Data’?
  Block2Data* _data2_;
  ^~~~~~~~~~
  Block5Data
emojier.c:5218:13: error: ‘Block2Data’ undeclared (first use in this function); did you mean ‘Block5Data’?
  _data2_ = (Block2Data*) _userdata_;
             ^~~~~~~~~~
             Block5Data
emojier.c:5218:24: error: expected expression before ‘)’ token
  _data2_ = (Block2Data*) _userdata_;
                        ^
In file included from /usr/lib64/glib-2.0/include/glibconfig.h:9,
                 from /usr/include/glib-2.0/glib/gtypes.h:32,
                 from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from emojier.c:26:
emojier.c:5219:41: error: request for member ‘_ref_count_’ in something not a structure or union
  if (g_atomic_int_dec_and_test (&_data2_->_ref_count_)) {
                                         ^~
/usr/include/glib-2.0/glib/gmacros.h:241:104: note: in definition of macro ‘G_STATIC_ASSERT’
 #define G_STATIC_ASSERT(expr) typedef char G_PASTE (_GStaticAssertCompileTimeAssertion_, __COUNTER__)[(expr) ? 1 : -1] G_GNUC_UNUSED
                                                                                                        ^~~~
emojier.c:5219:6: note: in expansion of macro ‘g_atomic_int_dec_and_test’
  if (g_atomic_int_dec_and_test (&_data2_->_ref_count_)) {
      ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib/gthread.h:32,
                 from /usr/include/glib-2.0/glib/gasyncqueue.h:32,
                 from /usr/include/glib-2.0/glib.h:32,
                 from emojier.c:26:
emojier.c:5219:41: error: request for member ‘_ref_count_’ in something not a structure or union
  if (g_atomic_int_dec_and_test (&_data2_->_ref_count_)) {
                                         ^~
/usr/include/glib-2.0/glib/gatomic.h:182:19: note: in definition of macro ‘g_atomic_int_dec_and_test’
     (void) (0 ? *(atomic) ^ *(atomic) : 1);                                  \
                   ^~~~~~
emojier.c:5219:41: error: request for member ‘_ref_count_’ in something not a structure or union
  if (g_atomic_int_dec_and_test (&_data2_->_ref_count_)) {
                                         ^~
/usr/include/glib-2.0/glib/gatomic.h:182:31: note: in definition of macro ‘g_atomic_int_dec_and_test’
     (void) (0 ? *(atomic) ^ *(atomic) : 1);                                  \
                               ^~~~~~
emojier.c:5219:41: error: request for member ‘_ref_count_’ in something not a structure or union
  if (g_atomic_int_dec_and_test (&_data2_->_ref_count_)) {
                                         ^~
/usr/include/glib-2.0/glib/gatomic.h:183:28: note: in definition of macro ‘g_atomic_int_dec_and_test’
     __sync_fetch_and_sub ((atomic), 1) == 1;                                 \
                            ^~~~~~
emojier.c:5221:17: error: request for member ‘self’ in something not a structure or union
   self = _data2_->self;
                 ^~
In file included from /usr/include/glib-2.0/glib.h:76,
                 from emojier.c:26:
/usr/include/glib-2.0/glib/gslice.h:66:24: error: expected expression before ‘)’ token
   else   (void) ((type*) 0 == (mem));     \
                        ^
emojier.c:5223:3: note: in expansion of macro ‘g_slice_free’
   g_slice_free (Block2Data, _data2_);
   ^~~~~~~~~~~~
emojier.c: At top level:
emojier.c:5228:33: error: unknown type name ‘Block2Data’; did you mean ‘Block5Data’?
 static gboolean _____lambda14_ (Block2Data* _data2_, GtkWidget* w, GdkEventButton* e) {
                                 ^~~~~~~~~~
                                 Block5Data
emojier.c: In function ‘______lambda14__gtk_widget_button_press_event’:
emojier.c:5252:11: warning: implicit declaration of function ‘_____lambda14_’; did you mean ‘_____lambda50_’? [-Wimplicit-function-declaration]
  result = _____lambda14_ (self, _sender, event);
           ^~~~~~~~~~~~~~
           _____lambda50_
emojier.c: At top level:
emojier.c:5287:33: error: unknown type name ‘Block2Data’; did you mean ‘Block5Data’?
 static gboolean _____lambda15_ (Block2Data* _data2_, GdkEventMotion* e) {
                                 ^~~~~~~~~~
                                 Block5Data
emojier.c: In function ‘______lambda15__gtk_widget_motion_notify_event’:
emojier.c:5382:11: warning: implicit declaration of function ‘_____lambda15_’; did you mean ‘_____lambda50_’? [-Wimplicit-function-declaration]
  result = _____lambda15_ (self, event);
           ^~~~~~~~~~~~~~
           _____lambda50_
emojier.c: In function ‘ibus_emojier_show_candidate_panel’:
emojier.c:5494:5: error: unknown type name ‘Block2Data’; did you mean ‘Block5Data’?
     Block2Data* _data2_;
     ^~~~~~~~~~
     Block5Data
In file included from /usr/include/glib-2.0/glib.h:76,
                 from emojier.c:26:
emojier.c:5531:29: error: ‘Block2Data’ undeclared (first use in this function); did you mean ‘Block5Data’?
     _data2_ = g_slice_new0 (Block2Data);
                             ^~~~~~~~~~
/usr/include/glib-2.0/glib/gslice.h:46:35: note: in definition of macro ‘g_slice_new0’
 #define  g_slice_new0(type)     ((type*) g_slice_alloc0 (sizeof (type)))
                                   ^~~~
/usr/include/glib-2.0/glib/gslice.h:46:40: error: expected expression before ‘)’ token
 #define  g_slice_new0(type)     ((type*) g_slice_alloc0 (sizeof (type)))
                                        ^
emojier.c:5531:15: note: in expansion of macro ‘g_slice_new0’
     _data2_ = g_slice_new0 (Block2Data);
               ^~~~~~~~~~~~
emojier.c:5532:12: error: request for member ‘_ref_count_’ in something not a structure or union
     _data2_->_ref_count_ = 1;
            ^~
emojier.c:5533:12: error: request for member ‘self’ in something not a structure or union
     _data2_->self = g_object_ref (self);
            ^~
emojier.c:5677:12: error: request for member ‘index’ in something not a structure or union
     _data2_->index = _tmp75_;
            ^~
emojier.c:5679:131: warning: implicit declaration of function ‘block2_data_ref’; did you mean ‘block5_data_ref’? [-Wimplicit-function-declaration]
     g_signal_connect_data ((GtkWidget*) _tmp76_, "button-press-event", (GCallback) ______lambda14__gtk_widget_button_press_event, block2_data_ref (_data2_), (GClosureNotify) block2_data_unref, 0);
                                                                                                                                   ^~~~~~~~~~~~~~~
                                                                                                                                   block5_data_ref
emojier.c:5679:131: warning: passing argument 4 of ‘g_signal_connect_data’ makes pointer from integer without a cast [-Wint-conversion]
     g_signal_connect_data ((GtkWidget*) _tmp76_, "button-press-event", (GCallback) ______lambda14__gtk_widget_button_press_event, block2_data_ref (_data2_), (GClosureNotify) block2_data_unref, 0);
                                                                                                                                   ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:28,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from emojier.c:27:
/usr/include/glib-2.0/gobject/gsignal.h:391:25: note: expected ‘gpointer’ {aka ‘void *’} but argument is of type ‘int’
             gpointer    data,
             ~~~~~~~~~~~~^~~~
emojier.c:5681:133: warning: passing argument 4 of ‘g_signal_connect_data’ makes pointer from integer without a cast [-Wint-conversion]
     g_signal_connect_data ((GtkWidget*) _tmp77_, "motion-notify-event", (GCallback) ______lambda15__gtk_widget_motion_notify_event, block2_data_ref (_data2_), (GClosureNotify) block2_data_unref, 0);
                                                                                                                                     ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:28,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from emojier.c:27:
/usr/include/glib-2.0/gobject/gsignal.h:391:25: note: expected ‘gpointer’ {aka ‘void *’} but argument is of type ‘int’
             gpointer    data,
             ~~~~~~~~~~~~^~~~
emojier.c: In function ‘ibus_emojier_update_favorite_emoji_dict’:
emojier.c:7543:14: warning: assignment to ‘GCompareFunc’ {aka ‘int (*)(const void *, const void *)’} from incompatible pointer type ‘int (*)(const char *, const char *)’ [-Wincompatible-pointer-types]
      _tmp46_ = g_strcmp0;
              ^
emojier.c:7559:30: warning: implicit declaration of function ‘NU’; did you mean ‘N_’? [-Wimplicit-function-declaration]
      (annotations == NULL) ? NU (annotations == NULL) ? NULL : (annotations = (_g_slist_free__g_free0_ (annotations), NULL));
                              ^~
                              N_
emojier.c:7559:125: error: expected ‘:’ before ‘;’ token
      (annotations == NULL) ? NU (annotations == NULL) ? NULL : (annotations = (_g_slist_free__g_free0_ (annotations), NULL));
                                                                                                                             ^
                                                                                                                             :
emojier.c:7567:13: warning: assignment to ‘GCompareFunc’ {aka ‘int (*)(const void *, const void *)’} from incompatible pointer type ‘int (*)(const char *, const char *)’ [-Wincompatible-pointer-types]
     _tmp58_ = g_strcmp0;
             ^
x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I../..  -I../../src -I../../src -include ../../config.h   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -pthread -I/usr/include/gio-unix-2.0/ -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -pthread -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -pthread -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/uuid -I/usr/include/freetype2 -I/usr/include/libdrm -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include  -DG_LOG_DOMAIN=\"IBUS\" -DBINDIR=\"/usr/bin\" -DPKGDATADIR=\"/usr/share/ibus\" -DIBUS_DISABLE_DEPRECATED -Wno-unused-variable -Wno-unused-but-set-variable -Wno-unused-function  -pthread -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include  -march=corei7 -O2 -pipe -c -o iconwidget.o iconwidget.c
emojier.c: In function ‘ibus_emojier_ewhite_label_real_get_preferred_height’:
emojier.c:9560:13: error: ‘_vala_naeight’ undeclared (first use in this function); did you mean ‘_vala_assert’?
   _tmp17_ = _vala_naeight = _tmp17_;
             ^~~~~~~~~~~~~
             _vala_assert
emojier.c: At top level:
emojier.c:10020:8: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘++’ token
  length++;
        ^~
emojier.c:10021:3: error: expected identifier or ‘(’ before ‘}’ token
   }
   ^
emojier.c:10022:2: error: expected identifier or ‘(’ before ‘}’ token
  }
  ^
emojier.c:10023:2: error: expected identifier or ‘(’ before ‘return’
  return length;
  ^~~~~~
emojier.c:10024:1: error: expected identifier or ‘(’ before ‘}’ token
 }
 ^
make[3]: *** [Makefile:847: emojier.o] Error 1
make[3]: *** Waiting for unfinished jobs....
rm ibus-emoji.7
make[3]: Leaving directory '/var/tmp/portage/app-i18n/ibus-1.5.19/work/ibus-1.5.19/ui/gtk3'
make[2]: *** [Makefile:513: all-recursive] Error 1
make[2]: Leaving directory '/var/tmp/portage/app-i18n/ibus-1.5.19/work/ibus-1.5.19/ui'
make[1]: *** [Makefile:687: all-recursive] Error 1
make[1]: Leaving directory '/var/tmp/portage/app-i18n/ibus-1.5.19/work/ibus-1.5.19'
make: *** [Makefile:594: all] Error 2
 * ERROR: app-i18n/ibus-1.5.19::gentoo failed (compile phase):
 *   emake failed
 * 
 * If you need support, post the output of `emerge --info '=app-i18n/ibus-1.5.19::gentoo'`,
 * the complete build log and the output of `emerge -pqv '=app-i18n/ibus-1.5.19::gentoo'`.
 * The complete build log is located at '/var/tmp/portage/app-i18n/ibus-1.5.19/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/app-i18n/ibus-1.5.19/temp/environment'.
 * Working directory: '/var/tmp/portage/app-i18n/ibus-1.5.19/work/ibus-1.5.19'
 * S: '/var/tmp/portage/app-i18n/ibus-1.5.19/work/ibus-1.5.19'

>>> Failed to emerge app-i18n/ibus-1.5.19, Log file:

>>>  '/var/tmp/portage/app-i18n/ibus-1.5.19/temp/build.log'

 * Messages for package app-i18n/ibus-1.5.19:

 * ERROR: app-i18n/ibus-1.5.19::gentoo failed (compile phase):
 *   emake failed
 * 
 * If you need support, post the output of `emerge --info '=app-i18n/ibus-1.5.19::gentoo'`,
 * the complete build log and the output of `emerge -pqv '=app-i18n/ibus-1.5.19::gentoo'`.
 * The complete build log is located at '/var/tmp/portage/app-i18n/ibus-1.5.19/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/app-i18n/ibus-1.5.19/temp/environment'.
 * Working directory: '/var/tmp/portage/app-i18n/ibus-1.5.19/work/ibus-1.5.19'
 * S: '/var/tmp/portage/app-i18n/ibus-1.5.19/work/ibus-1.5.19'
```

**Comment by fujiwarat** [comment]
Duplicate of #2042
